### PR TITLE
Add spot light with example

### PR DESCRIPTION
**Pull request by DunaevaOlga**
### 1. What has been done
A **SpotLight** class has been added that implements conical light in the **direction** and with focusing **p**.
### 2. Why we did it https://github.com/martinchristen/pyRT/issues/9
pyRT most powerful RT engine written in python. For needs in our project we did not find the spotlight feature. So spotlight has been implemented and introduced to the pyRT.

### 3. How we did it
To share several different light sources, a virtual function _intensity_ has been added, which calculates the intensity of a given light source at a point. The function is called in _simple.rt_ in the __shadow_ function. By default, the function returns 1, for SpotLight, the value is calculated depending on the distance from the point to the light source.
### 4. How to check
For using spot light call the _scene.addLight_ function with the _SpotLight_ parameter, for which you need to specify the position of the light source, direction and degree of focus. Also can be specify the parameters for calculating the intensity.
Example:
scene.addLight(SpotLight(Vec3(0, 0, 5), Vec3(-0.5, 0.5, -1), 7))

### CODE REVIEW

File: pyrt/light/pointlight.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def __init__(self, position: Vec3) -> None:
         Not yet complete - params will change in near future!
         :param position: position of the light
         """
-        Light.__init__(self, "PointLight")
+        Light.__init__(self,  "PointLight")
```
Comment (daniilsavchuk) [inline pyrt/light/pointlight.py]: Unnecessary change

File: pyrt/renderer/simplert.py [hunk 2]
```diff
@@ -50,17 +50,21 @@ def _shade(self, scene: Scene, ray: Ray, hitrecord: HitRecord) -> tuple:
...
+            fs += light.intensity(shadowray) 
+
             local_num_shadow_rays += 1
             for testelement in scene.nodes:
                 if testelement != hitrecord.obj:  # avoid self-intersection
                     if testelement.hitShadow(shadowray):
                         numshadow += 1
                         break
+
+        fs = np.clip(fs, 0, 1)
```
Comment (daniilsavchuk) [inline pyrt/renderer/simplert.py:67]: Was it been tested with point light again?

**Review by daniilsavchuk — COMMENTED** [review]
I've put some useful comments here. Thank you!

File: pyrt/light/pointlight.py (hunk 1 above)
Comment (DunaevaOlga) [reply pyrt/light/pointlight.py]: Fixed

File: pyrt/renderer/simplert.py (hunk 2 above)
Comment (DunaevaOlga) [reply pyrt/renderer/simplert.py:67]: Yes
